packages/common/src/types/Range.ts

```diff
@@ -12,29 +12,24 @@ export class Range {
   readonly end: Position;
 
   /**
-   * Create a new range from two positions. If `start` is not
-   * before or equal to `end`, the values will be swapped.
+   * Create a new range from two positions.
+   * The earlier of `p1` and `p2` will be used as the start position.
    *
-   * @param start A position.
-   * @param end A position.
+   * @param p1 A position.
+   * @param p2 A position.
    */
-  constructor(start: Position, end: Position);
+  constructor(p1: Position, p2: Position);
 
   /**
-   * Create a new range from number coordinates. It is a shorter equivalent of
-   * using `new Range(new Position(startLine, startCharacter), new Position(endLine, endCharacter))`
+   * Create a new range from number coordinates.
+   * It is equivalent to `new Range(new Position(line1, char1), new Position(line2, char2))`
    *
-   * @param startLine A zero-based line value.
-   * @param startCharacter A zero-based character value.
-   * @param endLine A zero-based line value.
-   * @param endCharacter A zero-based character value.
+   * @param line1 A zero-based line value.
+   * @param char1 A zero-based character value.
+   * @param line2 A zero-based line value.
+   * @param char2 A zero-based character value.
    */
-  constructor(
-    startLine: number,
-    startCharacter: number,
-    endLine: number,
-    endCharacter: number,
-  );
+  constructor(line1: number, char1: number, line2: number, char2: number);
 
   constructor(...args: any[]) {
     const [start, end]: [Position, Position] = (() => {
```

Comment:
I think we should prob mirror the naming from the function signature?

Reply:
done